db/c.cc

```diff
@@ -1057,6 +1057,35 @@ rocksdb_column_family_handle_t* rocksdb_create_column_family(
   return handle;
 }
 
+rocksdb_column_family_handle_t** rocksdb_create_column_families(
+    rocksdb_t* db, const rocksdb_options_t* column_family_options,
+    int num_column_families, const char* const* column_family_names,
+    char** errptr) {
+  std::vector<ColumnFamilyHandle*> handles;
+  std::vector<std::string> names;
+  for (int i = 0; i != num_column_families; ++i) {
+    names.push_back(std::string(column_family_names[i]));
+  }
+  SaveError(errptr, db->rep->CreateColumnFamilies(
+                        ColumnFamilyOptions(column_family_options->rep), names,
+                        &handles));
+
+  rocksdb_column_family_handle_t** c_handles =
+      static_cast<rocksdb_column_family_handle_t**>(malloc(
+          sizeof(rocksdb_column_family_handle_t*) * num_column_families));
+  for (int i = 0; i != num_column_families; ++i) {
```

Comment:
`handles` is not guaranteed to have size `num_column_families` on error: https://github.com/facebook/rocksdb/blob/ca50ccc71a1ce89008b4737e74f321b8df8a3b5b/include/rocksdb/db.h#L357-L359

Should we iterate up to `handles.size()` instead? Although, that introduces a problem that the `rocksdb_create_column_families()` caller does not know the number of column families that were created. Perhaps you can add a `size_t*` argument to mitigate that, similar to `rocksdb_list_column_families()`.

Reply:
Thanks, I had missed that detail.